timeutil/schedule_test.go

```diff
@@ -762,3 +762,141 @@ func (ts *timeutilSuite) TestScheduleNext(c *C) {
+		}, {
+			// last Monday of the month, at 10:00
+			schedule: "mon5,10:00-11:00",
+			// first Monday of the month, 11:00, right after
+			// 'previous first Monday' run
+			now:       "2017-02-27 10:59:20",
+			expecting: true,
+		}, {
+			// from first Monday of the month to the second Tuesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon1-tue2,10:00-12:00",
+			// Thursday, 11:10
+			now:       "2017-02-09 11:10:00",
+			expecting: true,
+		}, {
+			// from first Monday of the month to the second Tuesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon1-tue2,10:00~12:00",
+			// Thursday, 11:10
+			now:       "2017-02-02 11:10:00",
+			expecting: false,
+		}, {
+			// from first Monday of the month to the second Tuesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon1-tue2,10:00~12:00",
+			// Monday, 11:10
+			now:       "2017-02-06 11:10:00",
+			expecting: true,
+		}, {
+			// from first Monday of the month to the second Tuesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon1-tue2,10:00~12:00",
+			// Thursday, 11:10
+			now:       "2017-02-16 11:10:00",
+			expecting: false,
+		}, {
+			// from first Monday of the month to the second Tuesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon1-tue2,10:00~12:00",
+			// Thursday, 11:10
+			now:       "2017-02-16 11:10:00",
+			expecting: false,
+		}, {
+			// from first Monday of the month to the second Tuesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon1-tue2,10:00~12:00",
+			// Thursday, 11:10
+			now:       "2017-02-09 11:10:00",
+			expecting: true,
+		}, {
+			// from first Tuesday of the month to the second Monday of
+			// the month, at 10:00 to 12:00
+			schedule: "tue1-mon2,10:00~12:00",
+			// Thursday, 11:10
+			now:       "2017-02-09 11:10:00",
+			expecting: true,
+		}, {
+			// from 4th Monday of the month to the last Wednesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon4-wed5,10:00~12:00",
+			// Schedule ends up being Feb 20 - Feb 22 2017
+			now:       "2017-03-01 11:10:00",
+			expecting: false,
+		}, {
+			// from 4th Monday of the month to the last Wednesday of
+			// the month, at 10:00 to 12:00
+			schedule: "mon4-wed5,10:00~12:00",
+			// Schedule ends up being Feb 20 - Feb 22 2017
+			now:       "2017-02-23 11:10:00",
+			expecting: false,
+		}, {
+			// from last Monday of the month to the second Tuesday of
+			// the month, at 10:00
+			schedule: "mon1-tue2,10:00~12:00",
+			// Sunday, 11:10
+			now:       "2017-02-05 11:10:00",
+			expecting: false,
+		}, {
+			// twice between 9am and 11am
+			schedule:  "9:00-11:00/2",
+			now:       "2017-02-06 10:30:00",
+			expecting: true,
+		}, {
+			schedule:  "9:00-10:00,10:00-11:00",
+			now:       "2017-02-06 10:30:00",
+			expecting: true,
+		}, {
+			// 2 ranges, reversed order in spec
+			schedule: "10:00~11:00,9:00-10:00",
+			// sometime between 10am and 11am
+			now:       "2017-02-06 9:30:00",
+			expecting: true,
+		},
```

Comment:
Thanks for good tests! Can you also please add a check for schedule "23:59-23:59" (forcing one minute to be added)?